compiler+runtime/src/cpp/jank/analyze/processor.cpp

```diff
@@ -2701,69 +2700,83 @@ namespace jank::analyze
                 object_source(item),
                 latest_expansion(macro_expansions));
             }
-            if(has_catch)
-            {
-              /* TODO: Note where the other catch is. */
-              return error::analyze_invalid_try("Only one 'catch' form may be supplied.",
-                                                object_source(item),
-                                                latest_expansion(macro_expansions));
-            }
             has_catch = true;
 
-            /* Verify we have (catch <sym> ...) */
+            /* Verify we have (catch cpp/type <sym> ...) */
             auto const catch_list(runtime::list(item));
             auto const catch_body_size(catch_list->count());
-            if(catch_body_size == 1)
+            if(catch_body_size < 3)
             {
               return error::analyze_invalid_try(
-                "A symbol is required after 'catch', which is used as the binding to "
-                "hold the exception value.",
+                "Catch clause requires a type, a symbol, and a body.",
                 object_source(item),
                 latest_expansion(macro_expansions));
             }
+            auto catch_it(catch_list->data.rest());
+            auto const catch_type_form(catch_it.first().unwrap());
+            catch_it = catch_it.rest();
+            auto const catch_sym_form(catch_it.first().unwrap());
+            auto const catch_type(analyze(catch_type_form, current_frame, position, fn_ctx, true));
+            if(catch_type.is_err() || catch_type.expect_ok()->kind != expression_kind::cpp_type)
+            {
+              return error::analyze_invalid_try(
+                       "An exception type required after 'catch'",
+                       object_source(item),
+                       error::note{
+                         "An exception type is required before this form.",
+                         object_source(catch_list->data.rest().rest().first().unwrap()),
+                       },
+                       latest_expansion(macro_expansions))
+                ->add_usage(read::parse::reparse_nth(item, 1));
+            }
 
-            auto const sym_obj(catch_list->data.rest().first().unwrap());
-            if(sym_obj->type != runtime::object_type::symbol)
+            if(catch_sym_form->type != runtime::object_type::symbol)
```

Comment:
The error for this (which you didn't change in this PR) is also missing an 'is'.

> A symbol is required after 'catch' ...